src/burnchains/bitcoin/indexer.rs

```diff
@@ -706,9 +706,18 @@ impl BurnchainIndexer for BitcoinIndexer {
 
     /// Get a vector of the stacks epochs. This notion of epochs is dependent on the burn block height.
     /// Valid epochs include stacks 1.0, stacks 2.0, stacks 2.05, and so on.
+    ///
+    /// Choose according to:
+    /// 1) Use the custom epochs defined on the underlying `BitcoinIndexerConfig`, if they exist.
+    /// 2) Use hard-coded static values, otherwise.
+    ///
+    /// It is an error (panic) to set custom epochs if running on `Mainnet`.
     fn get_stacks_epochs(&self) -> Vec<StacksEpoch> {
         match self.config.epochs {
-            Some(ref epochs) => epochs.clone(),
+            Some(ref epochs) => {
+                assert!(self.runtime.network_id != BitcoinNetworkType::Mainnet);
+                epochs.clone()
+            }
```

Comment:
This seems like the wrong place to panic -- the panic should occur earlier, when the configuration is first applied (as in somewhere in `testnet/stacks-node/src/config` or the `neon` run loop). I think leave this assertion in place, but just add another one that occurs earlier.

Reply:
Seconded.  Never put off until runtime that which can be handled at boot time.

Reply:
I put a check in `BitcoinRegtestController::with_burnchain`.